tests/network/provider_migration/libprovider.py

```
@@ -31,18 +40,31 @@ def __exit__(self, exc_type, exc_val, exc_tb):
-        vm = self._get_vm_by_name(name=vm_name)
+    def power_on_vm(self, vm_name: str) -> vim.VirtualMachine:
+        LOGGER.info(f"Powering on VM '{vm_name}'")
+        vm = self.get_vm_by_name(name=vm_name)
         vm.PowerOn()
-        self._check_for_vm_status(vm=vm, status=vim.VirtualMachine.PowerState.poweredOn)
-        self._check_for_ip_obtained(vm=vm)
+        try:
+            self._check_for_vm_status(vm=vm, status=vim.VirtualMachine.PowerState.poweredOn)
+            self._check_for_ip_obtained(vm=vm)
+        except TimeoutExpiredError:
+            LOGGER.error(f"Timeout while trying to power on VM '{vm_name}' or waiting for IP address assignment")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)